source/MRViewer/MRPalette.cpp

```diff
@@ -503,6 +503,21 @@ Color Palette::getColor( float val ) const
     return Color();
 }
 
+VertColors Palette::getVertColors( const VertScalars& values, const VertBitSet& region, const VertBitSet* valids ) const
+{
+    MR_TIMER
+
+    VertColors result(region.find_last() + 1, Color::transparent() );
+    BitSetParallelFor( region, [&result, &values, valids, this] ( VertId v )
+    {
+        if ( valids && !valids->test( v ) )
+            result[v] = getInvalidColor();
+        else
+            result[v] = getColor( std::clamp( getRelativePos( values[v] ), 0.f, 1.f ) );
+    } );
+    return result;
```

Comment:
```suggestion
    VertColors result(region.size(), getInvalidColor() );
    const auto& validRegion = valids ? *valids : region;
    BitSetParallelFor( validRegion, [&result, &values, this] ( VertId v )
    {
            result[v] = getColor( std::clamp( getRelativePos( values[v] ), 0.f, 1.f ) );
    } );
    return result;

Reply:
In that case we set invalid Color for verts that out of region.
This is different from getUVCoords.